java/dataset/src/main/cpp/jni_util.h

```diff
@@ -91,6 +93,24 @@ void ReleaseNativeRef(jlong ref) {
   delete retrieved_ptr;
 }
 
+const char kJavaErrorDetailTypeId[] = "arrow::dataset::jni::JavaErrorDetail";
```

Comment:
Should this be a `static` member inside `JavaErrorDetail`, or perhaps `extern` and defined in the `.cc`?
As is, it seems there will be symbol clashes if `jni_util.h` is included by several `.cc` files.

Reply:
I have moved the variable into `.cc`.